sympy/functions/combinatorial/factorials.py

```diff
@@ -1057,3 +1058,37 @@ def _eval_is_nonnegative(self):
                 return True
             elif k.is_even is False:
                 return  False
+
+class multinomial(CombinatorialFunction):
+    r"""Implementation of the multinomial coefficient.
+    """
+    @classmethod
+    def eval(cls, *k):
+        k = [sympify(x) for x in k]
+        if all(x.is_Integer for x in k):
+            return factorial(Add(*k))/Mul(*(factorial(x) for x in k))
+        else:
+            from sympy import gamma
+            return gamma(Add(*k) + 1)/Mul(*(gamma(x + 1) for x in k))
+
+    def _eval_rewrite_as_factorial(self, *args, **kwargs):
+        return factorial(Add(*args))/Mul(*(factorial(x) for x in args))
+
+    def _eval_rewrite_as_binomial(self, *args, **kwargs):
+        from sympy import binomial
+        return Mul(*[binomial(j,k) for j,k in zip([Add(*args[:i]) for i in range(1,len(args)+1)],args)])
+
+    def _eval_rewrite_as_gamma(self, *k, **kwargs):
+        from sympy import gamma
+        return gamma(Add(*k) + 1)/Mul(*(gamma(x + 1) for x in k))
+
+    def _eval_is_integer(self):
+        k = self.args
+        k = [sympify(x) for x in k]
```

Comment:
```suggestion
        k = [_sympify(x) for x in k]
```

Reply:
Actually, this line should be deleted because the args are already sanitized

Reply:
alright